torch/nn/_functions/rnn.py

```diff
@@ -70,7 +70,7 @@ def StackedRNN(inners, num_layers, lstm=False, dropout=0, train=True):
     num_directions = len(inners)
     total_layers = num_layers * num_directions
 
-    def forward(input, hidden, weight):
+    def forward(input, hidden, weight, batch_sizes):
```

Comment:
@apaszke We need to confirm that these functions are not part of the public facing API. They're all defined and used solely in an underscore module, so I think this is the case.

Reply:
They're not, but why do we need to change the way we thread batch_sizes through those functions? This should be invisible to the tracer anyway

Reply:
they need to be visible to the tracer. How can the exported ONNX model run if it doesn't have access to lengths/batch_sizes? It's the same point as I made above - semantically, `(input, lengths)` is actually a single input.